### ISSUE TITLE
@param with variable name that ends with a dot silently turns off typechecking

### ISSUE THREAD

**Issue by evanadams-zz** [google/closure-compiler#499]
I was experimenting with an externs file and type checking.

I create an externs file with an event class with an addListener function. As you might expect, the addListener method takes a callback function as a param.

My goal was to then have a JS file that calls addListener() with a function of the wrong type.

I expected to get an error message, but was not getting one, even when the externs file specified a param of one of my classes and the JS code was passing in a function taking a {string}.

I chased the problem down to the addListener's JS-doc within the externs file.  It was:

```
/**
 * @param {function(!bug.MyClass): void} callback.
 */
AnotherClass.prototype.addListener = function(callback) {};
```

Took me quite a while to notice that the param name in the at-param statement is "callback." (contains a trailing dot) so it does not match the name in the function signature.

Normally, when a param is not mentioned in the at-param list, for example:

```
/**
 * @param {number} param2
 */
ewa.prototype.func1 = function(param) {
  console.log(param);
};

```

I see something like:

```
ewa.js:25: ERROR - parameter param2 does not appear in ewa.prototype.func1's parameter list
ewa.prototype.func1 = function(param) {

```

but there was no such message when the param name ended in a dot. 

And this had the effect of turning off all type checking.

I played around a bit and ending with a '*' does the same as a '.', but ending in a '-' or '+' generates an error message.
